controllers/admins.js

```diff
@@ -0,0 +1,34 @@
+const fs = require('fs');
+const admins = require('../data/admins.json');
+
+const getAdmins = (req, res) => {
+	res.status(200).json(admins)
+}
+
+const updateAdmin = (req, res) => {
+	const admin = admins.some(admin => admin.id === parseInt(req.params.id));
+	if(admin){
+		admins.map(admin => {
+			if (admin.id === parseInt(req.params.id)) {
+				!req.query.email ? null : admin.email = req.query.email;
+				!req.query.password ? null : admin.password = req.query.password;
+				!req.query.name ? null : admin.name = req.query.name;
+			}
+		})
+	}else {
+		res.status(400).json({ msg: `Admin with id: ${req.params.id} was not found` })
+	}
+		
+	fs.writeFile('./data/admins.json', JSON.stringify(admins), {}, (error) => {
+		if (error) {
+			return res.status(400).send(error)
+		}else {
+			return res.status(200).json({ msg: 'Admin updated', admins })
+		}
+	})
+}
```

Comment:
```
const updateAdmin = (req, res) => {
	let updatedAdmin
  const updatedAdmins = admins.map(admin => {
    if (admin.id === parseInt(req.params.id)) {
      updatedAdmin = admin
      return {
        email: req.query.email || admin.email,
        password: req.query.password || admin.password,
        name: req.query.name || admin.name,
      }
    }
    return admin
  })
  
  if(!updatedAdmin) {
    return res.status(400).json({ msg: `Admin with id: ${req.params.id} was not found` })
  }

	fs.writeFile('./data/admins.json', JSON.stringify(updatedAdmins), {}, (error) => {
		if (error) {
			return res.status(400).send(error)
		}
    return res.status(200).json({ msg: 'Admin updated', admins })
	})
}
```

Reply:
Hi Quito! I'm wondering, the way that i've done it, it's complety wrong? 
The part of map i know that is wrong, becasue im not return it.